### PR TITLE
add authentication part and add new sdk sample

### PR DESCRIPTION
**Pull request by xboxeer**
Fix issue: https://github.com/Azure/azure-sdk-for-net/issues/29901

### CODE REVIEW

**Comment by scottaddie** [comment]
@xboxeer I'd suggest updating the code sample in the source code repo instead. If a new version of the Track 1 library is ever released, these docs changes will be overwritten. @danieljurek, Do you agree?

**Comment by scottaddie** [comment]
Disregard my concern regarding overwriting. I've discussed with Daniel. Reviewing the PR now.

File: api/overview/azure/virtualmachines.md [hunk 1]
```diff
@@ -20,29 +20,193 @@ Create, configure, and scale out Windows and Linux virtual machines in Azure fro
...
+string computeName = "computeName";
+string admin = "admin";
+string pwd = "pwd";
+
+ArmClient client = new ArmClient(new DefaultAzureCredential());
+ResourceGroupResource resourceGroup = client.GetDefaultSubscription().GetResourceGroup(resourceGroupName);
+VirtualMachineCollection vms=resourceGroup.GetVirtualMachines();
+NetworkInterfaceCollection nics=resourceGroup.GetNetworkInterfaces();
+VirtualNetworkCollection vns=resourceGroup.GetVirtualNetworks();
+PublicIPAddressCollection publicIps=resourceGroup.GetPublicIPAddresses();
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:102]:
```suggestion
VirtualMachineCollection vms = resourceGroup.GetVirtualMachines();
NetworkInterfaceCollection nics = resourceGroup.GetNetworkInterfaces();
VirtualNetworkCollection vns = resourceGroup.GetVirtualNetworks();
PublicIPAddressCollection publicIps = resourceGroup.GetPublicIPAddresses();
```

File: api/overview/azure/virtualmachines.md [hunk 2]
```diff
@@ -20,29 +20,193 @@ Create, configure, and scale out Windows and Linux virtual machines in Azure fro
...
+        Location=AzureLocation.WestUS
+    }).Value;
+
+VirtualNetworkResource vnetResrouce = vns.CreateOrUpdate(
+    WaitUntil.Completed,
+    virtualNetworkName,
+    new VirtualNetworkData()
+    {
+        Location=AzureLocation.WestUS,
+        Subnets=
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:120]:
```suggestion
        Location = AzureLocation.WestUS,
        Subnets =
```

File: api/overview/azure/virtualmachines.md [hunk 3]
```diff
@@ -20,29 +20,193 @@ Create, configure, and scale out Windows and Linux virtual machines in Azure fro
...
+    {
+        new SubnetData()
+        {
+            Name="testSubNet",
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:124]:
```suggestion
            Name = "testSubNet",
```

File: api/overview/azure/virtualmachines.md [hunk 4]
```diff
@@ -20,29 +20,193 @@ Create, configure, and scale out Windows and Linux virtual machines in Azure fro
...
+            AddressPrefix = "10.0.0.0/24"
+        }
+    },
+        AddressPrefixes=
+    {
+        "10.0.0.0/16"
+    },
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:131]:
```suggestion
        {
            "10.0.0.0/16"
        },
```

File: api/overview/azure/virtualmachines.md [hunk 5]
```diff
@@ -20,29 +20,193 @@ Create, configure, and scale out Windows and Linux virtual machines in Azure fro
...
+    }).Value;
+
+NetworkInterfaceResource nicResource = nics.CreateOrUpdate(
+    WaitUntil.Completed,
+    networkInterfaceName,
+    new NetworkInterfaceData()
+    {
+        Location=AzureLocation.WestUS,
+        IPConfigurations={
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:140]:
```suggestion
        Location = AzureLocation.WestUS,
        IPConfigurations = {
```

File: api/overview/azure/virtualmachines.md [hunk 6]
```diff
@@ -20,29 +20,193 @@ Create, configure, and scale out Windows and Linux virtual machines in Azure fro
...
+        },
+        StorageProfile=new VirtualMachineStorageProfile()
+        {
+            OSDisk= new VirtualMachineOSDisk(DiskCreateOptionType.FromImage),
+            ImageReference=new ImageReference()
+            {
+                Offer="UbuntuServer",
+                Publisher="Canonical",
+                Sku="18.04-LTS",
+                Version="latest"
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:176]:
```suggestion
        HardwareProfile = new VirtualMachineHardwareProfile()
        {
            VmSize = VirtualMachineSizeType.BasicA0
        },
        OSProfile = new VirtualMachineOSProfile()
        {
            ComputerName = computeName,
            AdminUsername = admin,
            AdminPassword = pwd,
            LinuxConfiguration = new LinuxConfiguration()
            {
                DisablePasswordAuthentication = false,
                ProvisionVmAgent = true
            }
        },
        StorageProfile = new VirtualMachineStorageProfile()
        {
            OSDisk = new VirtualMachineOSDisk(DiskCreateOptionType.FromImage),
            ImageReference = new ImageReference()
            {
                Offer = "UbuntuServer",
                Publisher = "Canonical",
                Sku = "18.04-LTS",
                Version = "latest"
```

File: api/overview/azure/virtualmachines.md [hunk 7]
```diff
@@ -20,29 +20,193 @@ Create, configure, and scale out Windows and Linux virtual machines in Azure fro
...
+            }
+
+        },
+        NetworkProfile=new VirtualMachineNetworkProfile()
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:180]:
```suggestion
        NetworkProfile = new VirtualMachineNetworkProfile()
```

File: api/overview/azure/virtualmachines.md [hunk 8]
```diff
@@ -20,29 +20,193 @@ Create, configure, and scale out Windows and Linux virtual machines in Azure fro
...
+        {
+            NetworkInterfaces=
+        {
+            new VirtualMachineNetworkInterfaceReference()
+            {
+                Id=nicResource.Id
+            }
+        }
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:188]:
```suggestion
            NetworkInterfaces =
            {
                new VirtualMachineNetworkInterfaceReference()
                {
                    Id = nicResource.Id
                }
            }
```

**Review by scottaddie — CHANGES_REQUESTED** [review]

**Comment by xboxeer** [comment]
@scottaddie updated according to your suggestion

File: api/overview/azure/virtualmachines.md [hunk 9]
```diff
@@ -18,31 +18,199 @@ To get started with Azure virtual machines, see [Create a Linux virtual machine
...
+    {
+        Location = AzureLocation.WestUS,
+        Subnets =
+    {
+        new SubnetData()
+        {
+            Name = "testSubNet",
+            AddressPrefix = "10.0.0.0/24"
+        }
+    },
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:127]:
```suggestion
        {
            new SubnetData()
            {
                Name = "testSubNet",
                AddressPrefix = "10.0.0.0/24"
            }
        },
```

File: api/overview/azure/virtualmachines.md [hunk 10]
```diff
@@ -18,31 +18,199 @@ To get started with Azure virtual machines, see [Create a Linux virtual machine
...
+        {
+            OSDisk = new VirtualMachineOSDisk(DiskCreateOptionType.FromImage),
+            ImageReference = new ImageReference()
+            {
+                Offer = "UbuntuServer",
+                Publisher = "Canonical",
+                Sku = "18.04-LTS",
+                Version = "latest"
+            }
+
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:182]: Remove blank line
```suggestion
```

File: api/overview/azure/virtualmachines.md [hunk 11]
```diff
@@ -18,31 +18,199 @@ To get started with Azure virtual machines, see [Create a Linux virtual machine
...
+        IPConfigurations = {
+        new NetworkInterfaceIPConfigurationData()
+        {
+            Name = "Primary",
+            Primary = true,
+            Subnet = new SubnetData() { Id = vnetResrouce?.Data.Subnets.First().Id },
+            PrivateIPAllocationMethod = NetworkIPAllocationMethod.Dynamic,
+            PublicIPAddress = new PublicIPAddressData() { Id = ipResource?.Data.Id }
+        }
+    }
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:149]: Fix indentation
```suggestion
            new NetworkInterfaceIPConfigurationData()
            {
                Name = "Primary",
                Primary = true,
                Subnet = new SubnetData() { Id = vnetResrouce?.Data.Subnets.First().Id },
                PrivateIPAllocationMethod = NetworkIPAllocationMethod.Dynamic,
                PublicIPAddress = new PublicIPAddressData() { Id = ipResource?.Data.Id }
            }
        }
```

File: api/overview/azure/virtualmachines.md [hunk 12]
```diff
@@ -18,31 +18,199 @@ To get started with Azure virtual machines, see [Create a Linux virtual machine
...
+        },
+    }).Value;
+
+NetworkInterfaceResource nicResource = nics.CreateOrUpdate(
+    WaitUntil.Completed,
+    networkInterfaceName,
+    new NetworkInterfaceData()
+    {
+        Location = AzureLocation.WestUS,
+        IPConfigurations = {
```
Comment (scottaddie) [inline api/overview/azure/virtualmachines.md:140]:
```suggestion
        IPConfigurations =
        {
```

**Review by scottaddie — APPROVED** [review]
Here are some suggestions to consider, and then this PR looks good.

**Pull request merged**